src/java.base/share/classes/java/net/HttpCookie.java

```diff
@@ -644,7 +650,7 @@ public void setHttpOnly(boolean httpOnly) {
      *
      * @return  {@code true} if they domain-matches; {@code false} if not
      */
-    public static boolean domainMatches(String domain, String host) {
+    public static boolean domainMatches(@Nullable String domain, @Nullable String host) {
```

Comment:
I think we had this discussion a few times, but just because I notice it here again: while I agree that null can be passed, the function always returns false if either argument is null.
So, allowing null here might hide a problem in your code and will be rather hard to track down.
But I'm fine with this, if that's what you prefer here.

Reply:
I likewise don't love these. But I am always nervous to make a type non-null if users might be passing `null` today: Given the behavior of Kotlin (and we hope someday Java!), anyone who tries to pass a null value would get not only a compile warning/error but also a `NullPointerException` at runtime, thanks to the automatic null check that's inserted for non-null types.

Maybe someday I'll learn how to easily run all Google's tests with a modified JDK so that I can pass in a JDK whose `HttpCookie` class is tweaked to throw an exception for null inputs. (I guess I'd ideally want to cover Android, too, which may be harder....) Not that that would _prove_ that such a change is safe, but it would be evidence, and it at least has the potential to prove that it's _unsafe_.

I'll add a comment about this in the source.